### ISSUE TITLE
make container should target linux

### ISSUE THREAD

**Issue by r0fls** [kubernetes/kube-state-metrics#352]
When I run `make container` on my mac, I would expect the resulting docker image to be usable:

```
~/go/src/k8s.io/kube-state-metrics [master] 🚀 make container
cp -r * /var/folders/6c/1n3kjfy533530c2xlj_8r90c0000gq/T/tmp.GBNQSp83
GOOS=darwin GOARCH=amd64 CGO_ENABLED=0 go build -o /var/folders/6c/1n3kjfy533530c2xlj_8r90c0000gq/T/tmp.GBNQSp83/kube-state-metrics
docker build -t quay.io/coreos/kube-state-metrics-amd64:v1.2.0 /var/folders/6c/1n3kjfy533530c2xlj_8r90c0000gq/T/tmp.GBNQSp83
Sending build context to Docker daemon  88.37MB
Step 1/5 : FROM scratch
 ---> 
Step 2/5 : COPY kube-state-metrics /
 ---> Using cache
 ---> c9421874ce4e
Step 3/5 : VOLUME /tmp
 ---> Using cache
 ---> e887ca62544b
Step 4/5 : ENTRYPOINT ["/kube-state-metrics", "--port=8080"]
 ---> Using cache
 ---> 3a267e34b867
Step 5/5 : EXPOSE 8080
 ---> Using cache
 ---> 3459b9ca91d1
Successfully built 3459b9ca91d1
Successfully tagged quay.io/coreos/kube-state-metrics-amd64:v1.2.0
# Adding check for amd64
docker tag quay.io/coreos/kube-state-metrics-amd64:v1.2.0 quay.io/coreos/kube-state-metrics:v1.2.0
```

However, presumably because it's setting `GOOS=darwin` in the `go build` command, I get:

```
~/go/src/k8s.io/kube-state-metrics [master] 🚀 docker run -t quay.io/coreos/kube-state-metrics-amd64:v1.2.0 quay.io/coreos/kube-state-metrics:v1.2.0
standard_init_linux.go:195: exec user process caused "exec format error"
```

**Comment by r0fls** [comment]
For reference here is the relevant section of the helm makefile: https://github.com/kubernetes/helm/blob/master/Makefile#L74

**Comment by andyxning** [comment]
@r0fls 

You're right. See docker issue FYI: https://github.com/moby/moby/issues/27723.
TL;DR: This is because we add darwin binary to an scratch image.

For now, kube-state-metrics will use local build env to set up `GOOS` and `GOARCH` when building the binary. we need to enable the cross-build binaries to easily dev and test on mac or linux or other platforms. But as for images, we can only build and push linux binary.
